### PR TITLE
Issue94 darweshi amerson

### LINKED ISSUE #94: Add loading in UI
As a user i should be able to see a loading on the screen when we are trying fetch the data from the API.
reference - https://mui.com/material-ui/react-progress/

### PR DESCRIPTION
**Pull request by Damerson1**
Fixes #94

Review after comments:

What was changed:
A loading UI circle using Material-UI components was added whenever the program attempts to fetch data from the API. This is for user view.

Why:
There was not a clear sign of whether or not the program was working for the user view. Now the user knows that the program is attempting to grab data from the API, since there is now a loading icon shown.

How was it changed after review:

AddedCircularProgress from Material-UI.

Created a new file called Loading.jsx in, DroneWorld\frontend\src\components

File contains code that allows for the setup of a loading component.

Made changes in LandingPage.jsx to allow loading, imported loading, A function called isLoading was added to check if loading is true or false.

Also added to the useEffect function to manage loadstates before and after the fetch. This was done by setting isLoading to true when fetching starts and by setting it to false when the fetch is done. This allows the loading to show at the correct times. 

If isLoading is true, a Loading component is created and shown. Otherwise, the page is displayed as usual. 

Updated versions to match the currently used versions for the project*
Removed changes to store file*
Removed changes to Home file.
Rereviewed changes listed, more below. 

Note: Many of the changes listed are not changes in code after crosschecking. I am unsure as to why the code, mainly in "LandingPage.jsx" is listing so many changes. The only changes I made are those mentioned here, the rest of the code is untouched to my knowledge.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -4,13 +4,14 @@
   "private": true,
   "dependencies": {
     "@date-io/date-fns": "2.16.0",
-    "@emotion/react": "^11.10.5",
-    "@emotion/styled": "^11.10.5",
+    "@emotion/react": "^11.13.3",
```
Comment (nguzinski) [inline frontend/package.json:7]: unsure about updating the version here

File: frontend/package.json [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
+    "@emotion/styled": "^11.13.0",
     "@mui/icons-material": "^5.11.0",
     "@mui/lab": "5.0.0-alpha.118",
-    "@mui/material": "^5.11.5",
+    "@mui/material": "^5.16.7",
     "@mui/styles": "5.11.2",
```
Comment (nguzinski) [inline frontend/package.json:12]: here as well

File: frontend/src/components/Loading.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+
+export default function Loading() {
+  return (
+    <Box sx={{ width: '100%', display: 'flex', justifyContent: 'center' }}>
+      <CircularProgress />
+    </Box>
+  );
+}
```
Comment (nguzinski) [inline frontend/src/components/Loading.js:11]: LGTM

**Review by nguzinski — COMMENTED** [review]
Looks good other than version change

File: frontend/src/LandingPage.jsx [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 import { useState } from 'react';
 import { Link } from 'react-router-dom';
 import { makeStyles } from '@mui/styles';
 import Box from '@mui/material/Box';
 import Button from '@mui/material/Button';
 import Modal from '@mui/material/Modal';
-import Typography from '@mui/material/Typography'; 
-import ReportDashboard from './components/ReportDashboard'; 
+import Typography from '@mui/material/Typography';
+import ReportDashboard from './components/ReportDashboard';
```
Comment (kate-holdener) [inline frontend/src/LandingPage.jsx:9]: What changed in the above 2 lines of code? I don't see it

File: frontend/src/pages/Home.jsx [hunk 5]
```diff
@@ -221,4 +221,4 @@ reliability, and efficiency of sUAS applications.
 };
 
 export default Home;
-             
+
```
Comment (kate-holdener) [inline frontend/src/pages/Home.jsx:224]: Looks like an accidental change, please remove

File: frontend/src/LandingPage.jsx (hunk 4 above)
Comment (Damerson1) [reply frontend/src/LandingPage.jsx:9]: Sorry for the late reply, there are no changes in this code here. I am unsure why it shows there are some.

File: frontend/src/LandingPage.jsx [hunk 6]
```diff
@@ -127,61 +131,61 @@ export default function LandingPage() {
...
-            sx={{
-              color: 'white',
-              padding: '15px 30px',
-              borderRadius: '10px',
-            }}
-          >
-            Create Simulation
-          </Button>
-        </Link>
+      {isLoading ? (
```
Comment (Damerson1) [inline frontend/src/LandingPage.jsx:150]: Only change here is that isLoading ? , was added in order to add loading functionality.

**Review by mohamdlog — CHANGES_REQUESTED** [review]
I noticed that the Loading.js file contains JSX code but is named with a .js extension. For consistency with the rest of the project, which uses .jsx for components with JSX, please rename it to Loading.jsx. This will help maintain uniformity in the codebase. Otherwise, the code looks good!

**Review by mohamdlog — CHANGES_REQUESTED** [review]
It seems you have accidentally duplicated Loading.js/Loading.jsx.

**Review by mohamdlog — APPROVED** [review]
Looks good. Any spaces or newlines that get added/removed are considered changes, which is why you may see differences that aren't visually noticeable

**Pull request merged**